liquidity/liquidity.go

```diff
@@ -457,6 +504,136 @@ func (m *Manager) ForceAutoLoop(ctx context.Context) error {
+	if err != nil {
+		return err
+	}
+
+	_, err = m.checkSummaryInflight(summary)
+	if err != nil {
+		return err
+	}
+
+	// Get all channels in order to calculate current hot wallet balance.
+	channels, err := m.cfg.Lnd.Client.ListChannels(ctx, false, false)
+	if err != nil {
+		return err
+	}
+
+	hotWalletBalance := btcutil.Amount(0)
+	for _, channel := range channels {
+		hotWalletBalance += channel.LocalBalance
+	}
+
+	// Since we're only autolooping-out we need to check if we are below
+	// the target, meaning that we already meet the requirements.
+	if hotWalletBalance <= m.params.EasyAutoloopTarget {
+		log.Debugf("hot wallet balance %v below target %v",
+			hotWalletBalance, m.params.EasyAutoloopTarget)
+		return nil
+	}
+
+	restrictions, err := m.cfg.Restrictions(ctx, swap.TypeOut)
+	if err != nil {
+		return err
+	}
+
+	// Calculate the amount that we want to loop out. Check against the max
+	// swap amount of terms.
+	amount := hotWalletBalance - m.params.EasyAutoloopTarget
+	if amount > restrictions.Maximum {
+		amount = btcutil.Amount(restrictions.Maximum)
+	}
+
+	log.Debugf("easy autoloop: hot_wallet=%v, target=%v, "+
+		"attempting to loop out %v", hotWalletBalance,
+		m.params.EasyAutoloopTarget, amount)
+
+	// Start building that swap.
+	builder := newLoopOutBuilder(m.cfg)
+
+	channel := m.pickEasyAutoloopChannel(
+		channels, restrictions, loopOut, loopIn, amount,
+	)
+	if channel == nil {
+		return fmt.Errorf("no eligible channel for easy autoloop")
+	}
+
+	log.Debugf("easy autoloop: picked channel %v with local balance %v",
+		channel.ChannelID, channel.LocalBalance)
+
+	swapAmt, err := btcutil.NewAmount(
+		math.Min(channel.LocalBalance.ToBTC(), amount.ToBTC()),
+	)
+	if err != nil {
+		return err
+	}
+
+	// Override our current parameters in order to use the const percent
+	// limit of easy-autoloop.
+	easyParams := m.params
+	easyParams.FeeLimit = &FeePortion{
+		PartsPerMillion: defaultFeePPM,
+	}
+
+	// Set the swap outgoing channel to the chosen channel.
+	outgoing := []lnwire.ShortChannelID{
+		lnwire.NewShortChanIDFromInt(channel.ChannelID),
+	}
+
+	suggestion, err := builder.buildSwap(
+		ctx, channel.PubKeyBytes, outgoing, swapAmt, true, easyParams,
+	)
+	if err != nil {
+		return err
+	}
+
+	swap := loop.OutRequest{}
+
+	if t, ok := suggestion.(*loopOutSwapSuggestion); ok {
+		swap = t.OutRequest
+	} else {
+		return fmt.Errorf("unexpected swap suggestion type: %T", t)
+	}
+
+	// Dispatch a sticky loop out.
+	go m.dispatchStickyLoopOut(
+		ctx, swap, defaultAmountBackoffRetry, defaultAmountBackoff,
+	)
+
+	return nil
+}
+
+// autoCloseInactivePeers automatically closes channels with peers that are
+// considered inactive. This will check channel opens & closes, payment
+// forwards, and payments sent or received. This is mainly executed as a way to
+// free up liquidity that is locked towards a peer that is not good enough to
+// handle it.
+func (m *Manager) autoCloseInactivePeers(ctx context.Context) {
```

Comment:
Wrong PR?

Reply:
Left over from when this was within this PR's scope